src/mlpack/methods/hmm/hmm.hpp

```diff
@@ -290,6 +290,75 @@ class HMM
    */
   double LogLikelihood(const arma::mat& dataSeq) const;
 
+  /**
+   * Compute the log of the scaling factor of the given emission probability
+   * at time t. To calculate the log-likelihood for the whole sequence,
+   * accumulate log scale over the entire sequence
+   * This is meant for incremental or streaming computation of the
+   * log-likelihood of a sequence. For the first data point, provide an empty
+   * forwardLogProb vector.
+   *
+   * @param emissionLogProb emission probability at time t.
+   * probability up to time t-1
+   * @param forwardLogProb Vector in which forward probabilities will be saved.
+   * Passing forwardLogProb as an empty vector indicates the start of sequence
+   * or time t=0
+   * @return Log scale factor of the given sequence of emission at time t.
+   */
+  double EmissionLogScaleFactor(const arma::vec& emissionLogProb,
+                       arma::vec& forwardLogProb) const;
+  /**
+   * Compute the log-likelihood of the given emission probability up to time t,
+   * storing the result in logLikelihood.
+   * This is meant for incremental or streaming computation of the
+   * log-likelihood of a sequence. For the first data point, provide an empty
+   * forwardLogProb vector.
+   *
+   * @param emissionLogProb emission probability at time t.
+   * @param logLikelihood Log-likelihood of the given sequence of emission
+   * probability up to time t-1
```

Comment:
```suggestion
   * @param logLikelihood Log-likelihood of the given sequence of emission
   *     probability up to time t-1.  This will be overwritten with the log-likelihood
   *     of the given emission probability up to time t.
```